src/plugins/query_enhancements/public/plugin.tsx

```diff
@@ -117,6 +121,11 @@ export class QueryEnhancementsPlugin
       },
     });
 
+    const ASYNC_TRIGGER: Trigger<typeof ASYNC_TRIGGER_ID> = {
+      id: ASYNC_TRIGGER_ID,
+    };
+    uiActions.registerTrigger(ASYNC_TRIGGER);
```

Comment:
Piggy-backing off @ashwin-pc's comment, this PR sets up `queryEnhancements` so that after we have this trigger registered, we can use it to execute any progress updates, like when we add async stuff to PPL?